stravalib/client.py

```diff
@@ -686,13 +683,24 @@ def upload_activity(self, activity_file, data_type, name=None, description=None,
                               nordicski, alpineski, backcountryski, iceskate, inlineskate,
                               kitesurf, rollerski, windsurf, workout, snowboard, snowshoe
                               Type detected from file overrides, uses athlete's default type if not specified
+                              WARNING - This param is supported (as of 2022-11-15), but not
+                              documented and may be removed in the future.
         :type activity_type: str
 
-        :param private: (optional) set to True to mark the resulting activity as private, 'view_private' permissions will be necessary to view the activity
+        :param private: (optional) set to True to mark the resulting activity as private,
+                        'view_private' permissions will be necessary to view the activity.
```

Comment:
similar question to above - if it's not supported in the strava API does that mean it just no longer works? so we should remove it and add a warning to the user that this particular API feature has been removed by strava?

Reply:
`private` also doesn't exist anymore per the Strava API docs. But contrary to `activity_type`, it seems to be ignored by Strava: providing private=True/"true"/1/etc doesn't make the uploaded activity private.